ldap/servers/slapd/conntable.c

```diff
@@ -273,22 +263,21 @@ connection_table_get_connection(Connection_Table *ct, int sd)
 {
     PR_Lock(ct->table_mutex);
 
-    PR_ASSERT(ct->conn_next_offset != 0);
     Connection *c = ct->c_freelist[ct->conn_next_offset];
     if (c != NULL) {
         /* We allocated it, so now NULL the slot and move forward. */
-        ct->c_freelist[ct->conn_next_offset] = NULL;
-        /* Handle overflow. */
-        ct->conn_next_offset = (ct->conn_next_offset + 1) % ct->size;
-        if (ct->conn_next_offset == 0) {
-            /* Never use slot 0 */
-            ct->conn_next_offset += 1;
-        }
+        ct->c_freelist[ct->conn_next_offset++] = NULL;
```

Comment:
PR_ASSERT here that conn_next_offset must be less than < size + 1.

Reply:
Fixed